src/SlidingWindow.cpp

```diff
@@ -245,21 +315,53 @@ class SlidingWindowOnFunctionAndLoop : public IRMutator {
                 return stmt;
             }
 
+            string new_loop_min_name = unique_name('x');
+            Expr new_loop_min_var = Variable::make(Int(32), new_loop_min_name);
+            Expr new_loop_min_eq;
+            if (can_slide_up) {
+                new_loop_min_eq =
+                    substitute(loop_var, loop_min, min_required) == substitute(loop_var, new_loop_min_var, prev_max_plus_one);
+            } else {
+                new_loop_min_eq =
+                    substitute(loop_var, loop_min, max_required) == substitute(loop_var, new_loop_min_var, prev_min_minus_one);
+            }
+            // Ignore unsafe promises (intended for the ones generated by
+            // TailStrategy::GuardWithIf, but may be relevant in other cases).
+            new_loop_min_eq = lower_safe_promises(new_loop_min_eq);
```

Comment:
Would it be better to get the solver to look through promises in the same was as it looks through likely and likely_if_innermost?